fetch/mesh_data.go

```diff
@@ -126,14 +127,19 @@ func (f *Fetch) GetPoetProof(ctx context.Context, id types.Hash32) error {
 	case <-ctx.Done():
 		return ctx.Err()
 	case <-pm.completed:
-		if pm.err != nil {
-			f.logger.WithContext(ctx).With().Warning("failed to get hash",
-				log.String("hint", string(datastore.POETDB)),
-				log.Stringer("hash", id),
-				log.Err(pm.err))
-		}
 	}
-	return pm.err
+	switch {
+	case pm.err == nil:
+		return nil
+	case errors.Is(pm.err, activation.ErrObjectExists):
```

Comment:
please add comment on why this could happen. it's not a normal condition in the node implementation.
for other objects, we don't have fetching mechanism other than the hash fetcher.